source/docs/software/advanced-control/state-space/state-space-pose_state-estimators.rst

```diff
@@ -0,0 +1,39 @@
+WPILib Pose and State estimators
+================================
+
+WPILib includes pose and state estimators for differential, swerve and mecanum drivetrains. These estimators are designed to be drop-in replacements for the existing odometry classes, with added features that utilize Extended Kalman Filters to fuse latency-compensated robot pose estimates with encoders. In fact, if only ``update`` is called on these estimators, they will behave identically to their corrosponding odometry classes. These estimators can account for encoder drift and noisy vision data.
+
+Pose estimators estimate robot position using a state-space system with the states :math:`\begin{bmatrix}x & y & \theta \end{bmatrix}^T`, which can represent robot position as a ``Pose2d``. WPILib includes ``DifferentialDrivePoseEstimator``, ``SwerveDrivePoseEstimator`` and ``MecanumDrivePoseEstimators`` to estimate robot position. In these, users call ``update`` periodically with encoder and gyro measurements (same as the odometry classes) to update the robot's estimated position. When the robot receives measurements of it's field-relative position (encoded as a ``Pose2d``) from sensors such as computer vision or V-SLAM, the pose estimator latency-compensates the measurement to accurately estimate robot position. 
+
+The pose estimators perform latency compensation by storing a list of past observer states ,including estimated state :math:`\mathbf{\hat{x}}, error covariance :math:`\mathbf{P}`, inputs and local measurements. When new measurements are applied, the state of the estimator is rolled back to the its timestamp, the filter corrects its state estimate with the new measurement, and the filter applies the inputs between the measurement timestamp and now to incorporate the new measurement. This allows for vision solutions with framerates which might otherwise make them unusable be a viable solution for robot localization.
```

Comment:
I think it would be better if this sentence read, "When new measurements are applied, the state of the estimator is first rolled back to the measurement's timestamp. Then, the filter corrects its state estimate with the new measurement and applies the inputs between the measurement timestamp and the present time to incorporate the new measurement."